nowcasting_dataset/data_sources/gsp/gsp_data_source.py

```diff
@@ -124,35 +124,52 @@ def get_locations(self, t0_datetimes: pd.DatetimeIndex) -> Tuple[List[Number], L
         Returns: list of x and y locations
 
         """
-        # Pick a random GSP for each t0_datetime, and then grab
-        # their geographical location.
-        x_locations = []
-        y_locations = []
+        total_gsp_nan_count = self.gsp_power.isna().sum().sum()
+        if total_gsp_nan_count == 0:
 
-        # TODO: Issue 305: Speed up this function by removing this for loop?
-        for t0_dt in t0_datetimes:
+            # get random gps metadata
```

Comment:
Tiny typo :slightly_smiling_face: 

```suggestion
            # get random GSP metadata
```